### PR TITLE
Rename uniq `--ignore-case` long flag name to `--insensitive`

### PR DESCRIPTION
**Pull request by webbedspace**
# Description

`sort`, `sort-by`, `str contains` and `find` all have a -i flag that controls case-sensitivity. However, unlike `uniq`, the expanded version is called `--insensitive` for all of these commands. This changes `uniq` to use the same name. While this is a compat break, since long names aren't especially popular (existing primarily for self-documentation purposes), I consider this on the shallow end of the compat-break scale.

`uniq -i` remains exactly the same.

# User-Facing Changes

`uniq --ignore-case` → `uniq --insensitive` (compat break)

# Tests + Formatting

Don't forget to add tests that cover your changes.

Make sure you've run and fixed any issues with these commands:

- `cargo fmt --all -- --check` to check standard code formatting (`cargo fmt --all` applies these changes)
- `cargo clippy --workspace --features=extra -- -D warnings -D clippy::unwrap_used -A clippy::needless_collect` to check that you're using the standard code style
- `cargo test --workspace --features=extra` to check that all tests pass

# After Submitting

If your PR had any user-facing changes, update [the documentation](https://github.com/nushell/nushell.github.io) after the PR is merged, if necessary. This will help us keep the docs up to date.

### CODE REVIEW

File: crates/nu-command/tests/commands/uniq.rs [hunk 1]
```diff
@@ -230,3 +230,12 @@ fn uniq_simple_vals_strs() {
...
     assert_eq!(actual.out, expected.out);
 }
+
+#[test]
+fn uniq_insensitive_long_flag() {
+    let actual = nu!(
+        cwd: ".", pipeline(
+        "[aaa aAa bbb BBB] | uniq --insensitive | to nuon"
+    ));
+    assert_eq!(actual.out, "[aaa Bbb]");
```
Comment (sholderbach) [inline crates/nu-command/tests/commands/uniq.rs:240]: Failed due to typo here, but I honestly think this test is kind of redundant to the already existing example.

Comment (webbedspace) [reply crates/nu-command/tests/commands/uniq.rs]: Sorry, I was trying something here (relating to #7192) and forgot to remove it.

**Comment by fdncred** [comment]
If mine was the only vote, I'd change every existence of `--insensitive` to `--ignore-case` just because it's more intuitive to me.

**Comment by kubouch** [comment]
+1 for `--ignore-case`. `--insensitive` is a bit vague: Insensitive to what? We could also go all the way for `--case-insensitive`.

**Comment by webbedspace** [comment]
Your honour, I think the word "sensitive" by itself in a purely programming context pretty much *only* means "case-sensitive", does it not? So the word "insensitive" has the same meaning. (Meanwhile, we've got at least four commands with `--ignore-errors` as their expansion of `-i`, which is another "ignore" flag.)

**Comment by webbedspace** [comment]
> Some prior art:
> 
> GNU grep: `--ignore-case` [ripgrep (`rg`)](https://github.com/BurntSushi/ripgrep): `--ignore-case` [The Silver Searcher (`ag`)](https://github.com/ggreer/the_silver_searcher): `--ignore-case` [`fd`](https://github.com/sharkdp/fd): `--ignore-case`

These are a little unfair, since the middle two are obliged to copy the API of the first one…

Although, a proper manpage search reveals that `less`, `tree` and `wget` also use `--ignore-case`. I'll recognise that as a standard.

**Comment by fdncred** [comment]
@webbedspace one thing that would be helpful is standardizing on `--ignore-case` throughout our commands. I mention this in case you're interested in putting together a PR.

**Pull request closed without merging**